spec/mono/1.0.md

```diff
@@ -0,0 +1,61 @@
+Clients that use strings of equal length and box-drawing characters to produce a layout may find that the actual rendering does not match their expectation.
+
+*Rationale:* The idea of character-grid layouts is fundamentally incompatible with wide Unicode support.
+Letters from non-Latin scripts rarely fit into grid cells whose size is optimized for Latin letters.
+Furthermore, the existence of combining characters and grapheme clusters means that applications cannot depend on the equivalence of "one character = one grid cell" when calculating the width that a certain string occupies on the character grid.
+That's why monospace layouting is not enabled by default.
+In fact, VT6-compliant terminals may not offer monospace layouting at all.
+
+### 1.1. Dependencies to other modules
+
+VT6 servers MUST NOT agree to using this module unless they have already agreed to using any version of `vt6/term` on the same server connection.
+
+### 1.2. Monospace layouting
+
+We take the term **terminal** to refer to a terminal as defined in [`vt6/term1.0`](https://vt6.io/std/term/1.0/).
+When the value of the `mono.enabled` property is changed, the server SHALL append a paragraph separator (U+2029) to the terminal document, and move the terminal's cursor to point at the end of the terminal document, after that paragraph separator.
+After this operation, the terminal's cursor SHALL NOT move past this paragraph separator, even when the specification of another module would allow it.
+
+*Rationale:* This ensures that that the cursor coordinates (see below) are well-defined.
+
+All text that is added to the terminal document while `mono.enabled` is true SHALL be displayed to the user using a monospace font, that is, a font where at least all printable ASCII characters have the same width.
+
+## 2. Message types for `vt6/mono1.0`
+
+None.
+
+TODO consider adding message types for querying and setting the cursor coordinates (we do not want to have this as a property since it would generate a metric ton of `core.pub` messages); cursor coordinates can be defined as y = number of linebreaks since `mono.enabled` was set to true, x = number of chars since last linebreak
+
+## 3. Properties for `vt6/mono1.0`
+
+### 3.1. The `mono.enabled` property
+
+- Acceptable values: booleans
+- Can be set by client: yes
+
+The value of this property indicates whether the terminal is in monospace layouting mode, as described in section 1.2.
+
+When this property is set by a client using a `core.set` message, and its server connection is then closed, the property SHALL be reset to the previous value, unless the property's value has also been set on some other server connection after that `core.set` message.
+
+*Rationale:* When a client sets `mono.enabled`, prints some text and exits, the shell will then observe the previous value of `mono.enabled` when printing its prompt.
+
+TODO consider upstreaming this scoping logic into core
```

Comment:
Could it occur that two or more programs inside a single pipe (or just two or more programs that run at the same time on the same frame) compete in a way that e. g. one sets `mono.enabled` to true and expects it to be true until set otherwise, but the other one sets `mono.enabled` to false while the first one still expects it to be true?
This is not only a problem for `mono.enabled`, but is necessary to consider when upstreaming to core.

Reply:
Yes, there is a race condition here, but I don't think it's important in practice. `mono.enabled` defaults to `false`, so most clients that set it are going to set it to true.

Reply:
But the race condition could be bad the other way round, as in that a client expects the terminal to be fully unicode-capable while another client already set `mono.enabled` to true a long time ago.

Reply:
After short discussion, the actual problem here is that it should not be possible for a client to set `mono.enabled` *unless* their stdout is connected to the terminal. So this question ties into #19.